src/App.tsx

```diff
@@ -15,71 +20,124 @@ export const goods = [
+        const button:HTMLElement | null = element.querySelector('.button');
+
+        if (button && button.classList.contains('is-info')) {
+          button.classList.remove('is-info');
+          button.setAttribute('data-cy', 'AddButton');
+          button.textContent = '+';
+        }
+      }
+    });
+
+    this.setState({ selectedGood: null });
+  };
+
+  handleClick = (good: string) => {
+    const element:HTMLElement | null = document.getElementById(`${good}`);
+
+    if (element) {
+      if (element.classList.contains('has-background-success-light')) {
+        this.reset();
+      } else {
+        this.reset();
+
+        if (element) {
+          element.classList.add('has-background-success-light');
+          const button:HTMLElement | null = element.querySelector('.button');
+
+          if (button) {
+            goods.forEach(good => {
+              const overButton:HTMLElement | null = document.getElementById(`${good}`);
+
+              if (overButton) {
+                const otherButton:HTMLElement | null
+                = overButton.querySelector('.button');
+
+                if (otherButton) {
+                  otherButton.classList.remove('is-info');
+                  otherButton.setAttribute('data-cy', 'AddButton');
+                  otherButton.textContent = '+';
+                }
+              }
+            });
+
+            button.setAttribute('data-cy', 'RemoveButton');
+            button.classList.add('is-info');
+            button.textContent = '-';
+          }
+        }
+
+        this.setState({ selectedGood: good });
+      }
+    }
+  };
+
+  render() {
+    const { selectedGood } = this.state;
+
+    return (
+      <main className="section container">
+        {selectedGood ? (
+          <h1 className="title is-flex is-align-items-center">
+            {selectedGood}
+            {' '}
+            is selected
+
             <button
-              data-cy="AddButton"
+              data-cy="ClearButton"
               type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Garlic
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </main>
-);
+              className="delete ml-3"
+              onClick={this.reset}
+            />
+          </h1>
+        ) : (
+          <h1 className="title">
+            No goods selected
+          </h1>
+        )}
+
+        <table className="table">
+          <tbody>
+            {goods.map(good => (
+              <tr
+                data-cy="Good"
+                id={good}
+                key={good}
+                className={good === 'Jam'
+                  ? 'has-background-success-light'
+                  : ''}
+              >
+                <td>
+                  <button
+                    data-cy={good === 'Jam' ? 'RemoveButton' : 'AddButton'}
```

Comment:
here don't need to compare the hardcoded value 'Jam' instead we should compare the iterated item to the current selectedGood state to avoid errors:

```
 data-cy={good === selectedGood
                      ? 'RemoveButton'
                      : 'AddButton'}
```